src/CommunityToolkit.Maui/Essentials/FileFolderDialog.android.cs

```diff
@@ -0,0 +1,435 @@
+										or FileSelectionMode.FileSave
+										or FileSelectionMode.FileSaveRoot)
+			{
+				selectedFileName = inputText?.Text + "";
+				result = this.directory + '/' + selectedFileName;
+				autoResetEvent?.Set();
+			}
+			else
+			{
+				result = this.directory;
+				autoResetEvent?.Set();
+			}
+		});
+
+		dialogBuilder.SetNegativeButton("Cancel", (_, _) => { });
+		dirsDialog = dialogBuilder.Create();
+
+		if (dirsDialog != null)
+		{
+			dirsDialog.CancelEvent += (_, _) => { autoResetEvent?.Set(); };
+			dirsDialog.DismissEvent += (_, _) => { autoResetEvent?.Set(); };
+			// Show directory chooser dialog
+			autoResetEvent = new AutoResetEvent(false);
+			dirsDialog.Show();
+		}
+
+		await Task.Run(() => autoResetEvent?.WaitOne());
+
+		return result ?? throw new FileNotFoundException($"Could not locate {directoryFile}");
+	}
+
+
+	static bool CreateSubDir(string newDir)
+	{
+		var newDirFile = new File(newDir);
+		return !newDirFile.Exists() && newDirFile.Mkdir();
+	}
+
+	List<string> GetDirectories(string dir)
+	{
+		var directories = new List<string>();
+		try
+		{
+			var directoryFile = new File(dir);
+
+			// if directory is not the base sd card directory add ".." for going up one directory
+			if ((isRootSelectionMode || !directory.Equals(sdCardDirectory, StringComparison.Ordinal)) &&
+				!("/".Equals(directory, StringComparison.Ordinal)))
+			{
+				directories.Add("..");
+			}
+
+			Log.Debug("~~~~", "m_dir=" + directory);
+			if (!directoryFile.Exists() || !directoryFile.IsDirectory)
+			{
+				return directories;
+			}
+
+			foreach (var file in directoryFile.ListFiles() ?? Array.Empty<File>())
+			{
+				if (file.IsDirectory)
+				{
+					// Add "/" to directory names to identify them in the list
+					directories.Add(file.Name + "/");
+				}
+				else if (fileSelectionMode is FileSelectionMode.FileOpen
+												or FileSelectionMode.FileOpenRoot
+												or FileSelectionMode.FileSave
+												or FileSelectionMode.FileSaveRoot)
+				{
+					// Add file names to the list if we are doing a file save or file open operation
+					directories.Add(file.Name);
+				}
+			}
+		}
+		catch (Exception)
+		{
+			// ignored
+		}
+
+		directories.Sort();
+		return directories;
+	}
+
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//////                                   START DIALOG DEFINITION                                    //////
+	//////////////////////////////////////////////////////////////////////////////////////////////////////////
+	AlertDialog.Builder CreateDirectoryChooserDialog(string title,
+		List<string> listItems,
+		EventHandler<DialogClickEventArgs> onClickListener,
+		string defaultFileName = "default")
+	{
+		var dialogBuilder = new AlertDialog.Builder(context);
+		////////////////////////////////////////////////
+		// Create title text showing file select type // 
+		////////////////////////////////////////////////
```

Comment:
Let's trim this comment to one line. We don't use block comments like this anywhere else
```suggestion
		// Create title text showing file select type
```